library/core/src/ptr/const_ptr.rs

```diff
@@ -355,7 +355,8 @@ impl<T: ?Sized> *const T {
     ///
     /// If any of the following conditions are violated, the result is Undefined Behavior:
     ///
-    /// * The computed offset, `count * size_of::<T>()` bytes, must not overflow `isize`.
+    /// * The computed offset, `count * size_of::<T>()` bytes (using unbounded arithmetic),
+    ///   must fit in an `isize`.
```

Comment:
What is the most clear and concise way to say "convert `count` and `size_of::<T>()` from whatever their types are into unbounded mathematical integers, multiply those, and check if the result fits in the value range of `isize`"?

Reply:
FWIW, the docs on `offset_from` say

> the absolute distance between the pointers, in bytes, computed on mathematical integers (without "wrapping around"), cannot overflow an `isize`

so maybe something like this could work

> The offset in bytes (`count * size_of::<T>()`), computed on mathematical integers (without "wrapping around"), must fit in an `isize`.

Reply:
I think "mathematical multiplication" is the most intuitive version here -- good catch with the `offset_from` docs, I will follow that.